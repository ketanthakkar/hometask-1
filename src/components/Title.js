import React from 'react';
import '../styleModules/Title.css';

const Title = ({title}) => <p className = 'title'> {title} </p>;

export default Title;