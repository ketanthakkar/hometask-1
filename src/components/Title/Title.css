.title {
    color:palevioletred;
    font-size: 1.6em;
}