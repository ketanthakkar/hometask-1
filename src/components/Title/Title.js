import React from 'react';
import './Title.css';

const Title = ({title}) => <p className = 'title'> {title} </p>;

export default Title;