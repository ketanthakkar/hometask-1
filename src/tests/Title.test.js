import React from 'react';
import { shallow } from 'enzyme';
import Title from '../components/Title';

describe('Title', () => {
  it('shoud render Title component', () => {
    const render = shallow(
      <Title title = 'test' />
    );
    expect(render.find('p').hasClass('title')).toBe(true);
    expect(render).toMatchSnapshot();
  })

  it('should not modify props', () => {
    const title = 'test';
    const render = shallow(
      <Title title = {title} />
    );
    expect(title).toEqual('test');
    expect(render).toMatchSnapshot();
  })
})