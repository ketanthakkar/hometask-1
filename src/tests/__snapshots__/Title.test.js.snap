// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title shoud render Title component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Title
    title="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        " ",
        "test",
        " ",
      ],
      "className": "title",
    },
    "ref": null,
    "rendered": Array [
      " ",
      "test",
      " ",
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          "test",
          " ",
        ],
        "className": "title",
      },
      "ref": null,
      "rendered": Array [
        " ",
        "test",
        " ",
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Title should not modify props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Title
    title="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        " ",
        "test",
        " ",
      ],
      "className": "title",
    },
    "ref": null,
    "rendered": Array [
      " ",
      "test",
      " ",
    ],
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " ",
          "test",
          " ",
        ],
        "className": "title",
      },
      "ref": null,
      "rendered": Array [
        " ",
        "test",
        " ",
      ],
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
